### PR TITLE
feat: scss modules support complete

### PR DESCRIPTION
**Pull request by ni0ki**
Added clean SCSS modules / support to docz, jest, and the build process.
Updated the TextInput component to reflect the changes.

The styles are now properly scoped to their component, with no collision risk. The resulting CSS is embedded in the final JS file, and is injected at runtime.

### CODE REVIEW

**Comment by sericaia** [comment]
This looks good, thanks. The only thing I would change is to do 2 different PRs next time. One related wits SCSS and another for the TextInput ;)

File: jest.config.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module.exports = {
   roots: ['<rootDir>/src'],
   transform: {
+    '.+\\.(css|styl|less|sass|scss)$':
+      '<rootDir>/node_modules/jest-css-modules-transform',
```
Comment (pducolin) [inline jest.config.js]: Isn't enough to give it `jest-css-modules-transform` instead of the full path with `<rootDir>` ?

File: package.json [hunk 2]
```diff
@@ -26,19 +26,19 @@
     "check-lint": "eslint --ext tsx,ts,js,json .",
     "exec-docz-update": "node config/updateDoczBuild.js",
     "generate-icon-components": "node config/generator/icons/generateIcons.js",
-    "generate-colors-stylesheets": "node config/generator/colors/generateColorStylesheets.js",
-    "sass:watch": "yarn sass --watch src:src"
+    "generate-colors-stylesheets": "node config/generator/colors/generateColorStylesheets.js"
   },
   "sideEffects": false,
   "dependencies": {
+    "@babel/runtime": "^7.9.2",
```
Comment (pducolin) [inline package.json]: this one can be removed

File: src/components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -18,38 +17,28 @@ const classNameFromFeedbackType: Record<TextInputFeedbackType, string> = {
   warning: 'warning'
 };
 
-const getClassNameFromFeedbackType = (
-  feedbackType?: TextInputFeedbackType
-): string => {
+const getClassNameFromFeedbackType = (feedbackType?: TextInputFeedbackType) => {
```
Comment (pducolin) [inline src/components/TextInput/TextInput.tsx]: as Mariem pointed out in my PR, we can refactor this and use directly the prop instead of converting it to text - I will change it later

File: src/components/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -13,43 +12,27 @@ interface TextInputProps
...
-    feedbackText.length > 0
-  );
+const shouldShowFeedback = ({ feedbackText, feedbackType }: TextInputProps) => {
+  return feedbackText && feedbackType;
 };
 
 export const TextInput: React.FC<TextInputProps> = props => {
-  const className = getClassNameFromFeedbackType(props.feedbackType);
-  const { feedbackText, feedbackType, ...htmlInputProps } = props;
+  const styledClassName = props.feedbackType
```
Comment (maryiam) [inline src/components/TextInput/TextInput.tsx:20]: I don't find this nested ternary clear and easily understandable.
maybe 
```js
props.feedbackType && style[props.feedbackType] ? style[props.feedbackType] : ''
```

File: src/components/TextInput/TextInput.module.scss [hunk 5]
```diff
@@ -63,19 +63,19 @@
     &::-ms-input-placeholder {
```
Comment (maryiam) [inline src/components/TextInput/TextInput.module.scss:63]: We should have a linter for styles to avoid this kind of changes in PRs

File: jest.config.js (hunk 1 above)
Comment (maryiam) [reply jest.config.js:5]: yes, I think you can omit the `<rootDir>/node_modules/` part

File: jest.config.js [hunk 6]
```diff
@@ -14,7 +16,6 @@ module.exports = {
     '^@ions/(.*)$': '<rootDir>/src/ions/$1',
     '@ions': '<rootDir>/src/ions/index',
     '@colors': '<rootDir>/src/colors/colors.ts',
-    '@colors/*': '<rootDir>/src/colors/*',
-    '\\.(css|less)$': 'identity-obj-proxy'
```
Comment (maryiam) [inline jest.config.js:18]: If `identity-obj-proxy` is not used any more than maybe we can delete it from our dependencies?

File: package.json [hunk 7]
```diff
@@ -82,13 +85,14 @@
     "rollup-plugin-terser": "^5.1.2",
     "rollup-plugin-typescript2": "^0.24.3",
     "sass": "^1.26.3",
+    "sass-loader": "^8.0.2",
     "semantic-release": "^15.13.27",
     "style-loader": "^1.1.3",
```
Comment (maryiam) [inline package.json:90]: Is `style-loader` still needed?

File: package.json [hunk 8]
```diff
@@ -38,7 +37,7 @@
   },
   "devDependencies": {
     "@ampproject/rollup-plugin-closure-compiler": "^0.10.0",
-    "@babel/runtime": "^7.9.2",
+    "@callstack/react-theme-provider": "^2.1.0",
```
Comment (maryiam) [inline package.json:40]: Why is this needed?

File: src/typings.d.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+declare module '*.module.scss' {
```
Comment (maryiam) [inline src/typings.d.ts:1]: Isn't it enough to declare the module `*.module.scss`? Why is `className` needed?

File: src/typings.d.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+  interface ClassNames {
+    [className: string]: string;
+  }
+  const classNames: ClassNames;
```
Comment (maryiam) [inline src/typings.d.ts:5]: You don't need an interface here, you can use [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt)
```js
classNames: Record<string, string>
```

File: src/typings.d.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+  export = classNames;
```
Comment (maryiam) [inline src/typings.d.ts:6]: Or export default classNames?

File: jest.config.js (hunk 1 above)
Comment (ni0ki) [reply jest.config.js:5]: Will do

File: src/components/TextInput/TextInput.tsx (hunk 4 above)
Comment (ni0ki) [reply src/components/TextInput/TextInput.tsx:20]: Will do

File: jest.config.js (hunk 6 above)
Comment (ni0ki) [reply jest.config.js:18]: Sure, will do

File: package.json (hunk 7 above)
Comment (ni0ki) [reply package.json:90]: Nope, I'll remove it

File: package.json (hunk 8 above)
Comment (ni0ki) [reply package.json:40]: Because we mixed up things in this TextInput PR, namely, it was the original one to provide SASS support, which was needed for the theming we started, which relies on this library. So everything related to SASS / PostCSS / theming is based on this branch.

File: src/typings.d.ts (hunk 9 above)
Comment (ni0ki) [reply src/typings.d.ts:1]: Yes, it is enough. I was originally using another rollup-postcss plugin which was causing some TS problems, with the changes the className part is indeed not needed :+1:

File: src/typings.d.ts (hunk 11 above)
Comment (ni0ki) [reply src/typings.d.ts:6]: Will be deleted !

File: src/components/TextInput/TextInput.tsx (hunk 3 above)
Comment (ni0ki) [reply src/components/TextInput/TextInput.tsx]: I changed it !

File: package.json (hunk 8 above)
Comment (maryiam) [reply package.json:40]: Maybe remove temporarily then? :)

File: src/components/TextInput/TextInput.module.scss (hunk 5 above)
Comment (oclero) [reply src/components/TextInput/TextInput.module.scss:63]: I'm adding the necessary lines to .prettierrc in another PR.

Comment (maryiam) [reply src/components/TextInput/TextInput.module.scss:63]: That's cool :) Thanks
